pkg/command/command.go

```diff
@@ -111,12 +55,34 @@ func withCopyOfWorkTree(sourcePath *string, work func(string) (string, error)) (
 			fmt.Fprintf(os.Stderr, "Unable to remove path %q, returned error: %s", tmpPath, err)
 		}
 	}()
-
-	err = fileutils.CopyDir(*sourcePath, tmpPath, true, []string{})
+	err = fileutils.CopyDir(*path, tmpPath, true, []string{})
 	if err != nil {
 		return "", err
 	}
+	repo, err := git.LoadRepository(ctx, tmpPath, *providerType, *token)
+	if err != nil {
+		return "", fmt.Errorf("could not load repository: %w", err)
+	}
 
-	message, err := work(tmpPath)
-	return message, err
+	// Run Command
+	cancelCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	switch args[1] {
+	case "new":
+		newCommand := flag.NewFlagSet(args[1], flag.ExitOnError)
+		group := newCommand.String("group", "", "Main application group")
+		app := newCommand.String("app", "", "Name of the application")
+		tag := newCommand.String("tag", "", "Application version/tag to set")
+		err := newCommand.Parse(args[2:])
+		if err != nil {
+			return "", err
+		}
+		return NewCommand(cancelCtx, cfg, repo, *group, *app, *tag)
```

Comment:
Should we not be passing the config as a reference? It may grow large over time and should conceptually be read-only.